### ISSUE TITLE
In jupterbook in anaconda, there is no output when I use vispy.Should I install some other packages?

### ISSUE THREAD

**Issue by yileichen123** [vispy/vispy#1528]

**Comment by djhoese** [comment]
I'm assuming you mean `jupyter notebook`. It is possible older versions of vispy (even the master branch at this point) are not compatible with the newest versions of jupyter notebook. I believe all of the necessary packages are already listed as dependencies for vispy and should have been installed. If you are receiving an error message (either in the python output or your browsers console) then that is a different story.

**Comment by mpspells** [comment]
For what it's worth, we have had success using ipywidgets 6.0.1 with vispy 0.5.3, and ipywidgets 7.0.1 with vispy 0.6 prerelease (from source). There are a couple of caveats:

- You have to be very careful to install compatible versions of vispy, ipywidgets, widgetsnbextension, and notebook; I think installing with pip has the correct dependencies for ipywidgets, though.
- If installing these dependencies using pip, make sure to enable the widgetsnbextension (`jupyter nbextension enable --py --sys-prefix widgetsnbextension`).
- If you've previously installed various jupyter python kernels or other notebook extensions, it can be very easy for the old versions to accidentally overrule new versions you've installed. I'd advise double checking that the kernel spec, notebook extension, and all python packages are all coming from the location you're expecting.

**Comment by mpspells** [comment]
Correct, it's to get ipywidgets to work. I think the step is not needed if the packages are installed via conda (or, I believe I read that sufficiently modern notebook removes the need to install the extension, though that may not be compatible with ipywidgets 6.*).
